pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -184,16 +161,13 @@ public AuthorizationService(ServiceConfiguration conf, PulsarResources pulsarRes
         if (!this.conf.isAuthorizationEnabled()) {
             return CompletableFuture.completedFuture(true);
         }
-        if (provider != null) {
-            return provider.isSuperUser(role, authenticationData, conf).thenComposeAsync(isSuperUser -> {
-                if (isSuperUser) {
-                    return CompletableFuture.completedFuture(true);
-                } else {
-                    return provider.canProduceAsync(topicName, role, authenticationData);
-                }
-            });
-        }
-        return FutureUtil.failedFuture(new IllegalStateException("No authorization provider configured"));
+        return provider.isSuperUser(role, authenticationData, conf).thenComposeAsync(isSuperUser -> {
```

Comment:
Maybe there could we use specific ``executor`` instead of ``commonPool`` when ``thenComposeAsync``?

Reply:
This is a fair question to ask, but I think fixing it is out of the scope of this PR. We could definitely consider adding an executor to the `AuthorizationService`, but I think we should consider refactoring this class first.

In my view, the larger problem here is that the `AuthenticationService` is making composite authorization decisions about who can produce by determining that super user or that `provider.canProduceAsync` is sufficient. I think the proper abstraction would only delegate to the `AuthenticationProvider`.

Reply:
I agree with your consideration.

I think we need to hear from everyone or discuss it in another PR. :-)